### PR TITLE
Backend/209/login testing

### LINKED ISSUE #209: Add integration tests for custom OIDC logic
Description:
There should be an integration test to cover the custom Open ID Connect (OIDC) logic in the login process.
This is addressing one of the verbal comments from the Sprint 2 PR regarding the "no cover" statements in areas where we required a response from the login.gov server. 

AC: 

- [ ] Integration tests to cover the custom logic in login.py
- [ ] Unit test to check HttpOnly cookie

Tasks: 

- [ ] Add Mozilla OIDC test server to docker
- [ ] Write tests to cover the login process
  - [ ] Mock state and nonce values
  - [ ] Log in to test server and retrieve token
  - [ ] Mock the OIDC endpoint and add data to session
  - [ ] Use token to log in to app
  - [ ] Write unit test to verify HttpOnly Cookie
 - [ ] Code documentation and annotation of why system is the way it is

DoD: 
 - [ ] @lauraGgit has reviewed and signed off
 - [ ] All tests are passing
 - [ ] There are no linting errors
 
 Note: I am assigning 3 story points to this task, but want to note the risk that we are currently making some assumptions about how the Mozilla test server works. The assumptions are based on good research, so i'm fairly confident it won't be a problem, but if the test server doesn't work the way we expect it could add some time to getting this to work properly.

Supporting Documentation:

(please include any relevant log snippets/ files/ screen shots here.)

Open Questions:

- Does this belong as an acceptance criteria or definition of done?

### PR DESCRIPTION
**Pull request by carltonsmith**
Closes #209 

- 100% coverage on login.py
- tests HTTPOnly Cookie on auth_check

Note: Because of the way that the Django testing system works, we are unable to test for the HTTPOnly Cookie from the back end. There would need to be an actual HTTP server running the tests, and Django testing doesn't work that way.

### [AC] This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [ ] Vulnerability Scan Test (Zap) has no high/critcial CVEs
- [x] Linting Tests (ESLint for UI and Flake8 for Python)
- [x] Unit Tests ( including a code coverage test report)
- [ ] Accessibility Tests (Pa11y UI Only)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### [DoD] This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: tdrs-backend/tdpservice/users/api/login.py [hunk 1]
```diff
@@ -132,8 +132,8 @@ def get(self, request, *args, **kwargs):
 
                 return response_redirect(user, id_token)
 
-        except Exception as e:  # pragma: no cover
-            logger.info(f"Error attempting to login/registeruser:  {e} at...")
+        except Exception as e:
+            logger.exception(f"Error attempting to login/registeruser:  {e} at...")
```
Comment (RafterGit) [inline tdrs-backend/tdpservice/users/api/login.py:136]: Please change `Error attempting to login/registeruser` to `Error attempting to login/register user`

File: tdrs-backend/tdpservice/users/test/test_auth.py [hunk 2]
```diff
@@ -86,6 +88,302 @@ def test_login_fails_without_state(api_client):
...
+        "id_token": os.environ["MOCK_TOKEN"]
+    }
+    mock_decode = mocker.patch("tdpservice.users.api.login.jwt.decode")
+    decoded_token = {
+        "email": "test@example.com",
+        "email_verified": True,
+        "nonce": nonce,
+        "iss": "https://idp.int.identitysandbox.gov",
+        "sub": "b2d2d115-1d7e-4579-b9d6-f8e84f4f56ca",
+        "verified_at": 1577854800
```
Comment (RafterGit) [inline tdrs-backend/tdpservice/users/test/test_auth.py:112]: Whats the specific reason for picking this epoch stamp that converts to `Wednesday, January 1, 2020`. Adding a comment as to why we're picking this epoch stamp might help.

File: tdrs-backend/tdpservice/users/test/test_auth_check.py [hunk 3]
```diff
@@ -1,9 +1,13 @@
 """Test the authorization check."""
 
 import pytest
+import logging
```
Comment (RafterGit) [inline tdrs-backend/tdpservice/users/test/test_auth_check.py:4]: We don't seem to be using logging in `test_auth_check` other than setting it to `INFO`. Can we take it out?

File: tdrs-backend/docker-compose.yml [hunk 4]
```diff
@@ -23,6 +23,7 @@ services:
       - DJANGO_SETTINGS_MODULE=tdpservice.settings.local
       - DJANGO_CONFIGURATION=Local
       - JWT_KEY=${JWT_KEY}
+      - JWT_KEY_TEST=${JWT_KEY_TEST}
```
Comment (RafterGit) [inline tdrs-backend/docker-compose.yml:26]: I don't see `JWT_KEY_TEST` being used anywhere.

**Review by RafterGit — CHANGES_REQUESTED** [review]

File: tdrs-backend/tdpservice/users/test/test_auth.py (hunk 2 above)
Comment (carltonsmith) [reply tdrs-backend/tdpservice/users/test/test_auth.py:113]: No particular reason for this. It was just an example token that we grabbed to mock with. It doesn't make any difference for the mock.

Comment (RafterGit) [reply tdrs-backend/tdpservice/users/test/test_auth.py:113]: copy does this mean it'll break  on new years? Would it be feasible to have this set to lets ays 15min from the current system time?

Comment (carltonsmith) [reply tdrs-backend/tdpservice/users/test/test_auth.py:113]: No, it doesn't actually read the value

**Review by RafterGit — APPROVED** [review]
LGTM.

**Pull request merged**